xdsl/dialects/dlti.py

```diff
@@ -47,12 +75,25 @@ class DLTIEntryMap(ParametrizedAttribute, ABC):
     # In MLIR, this is a DataLayoutEntryInterface.
     entries: ArrayAttr[DataLayoutEntryAttr]
 
+    def __init__(
+        self,
+        contents: ArrayAttr[DataLayoutEntryAttr] | DictValueType,
+    ):
+        if not isa(contents, ArrayAttr[DataLayoutEntryAttr]):
+            assert isinstance(contents, Mapping)
```

Comment:
Does pyright complain if this line is removed?

Reply:
Yes, for whoever instantiates the object it reports `error: Expected 0 positional arguments` if this is removed

Reply:
Yup, `Argument type is partially unknown` without this

Reply:
Ah then it might be worth flipping the check and doing `if isinstance(contents, Mapping):` instead